lib/src/rules/unnecessary_this.dart

```diff
@@ -85,11 +85,12 @@ class _UnnecessaryThisVisitor extends ScopedVisitor {
       lookUpElement = DartTypeUtilities.getCanonicalElement(
           nameScope.lookup(parent.methodName, definingLibrary));
       localElement = parent.methodName.bestElement;
-    } else {
+    }
+    // Error in code
+    if (localElement == null) {
       return null;
     }
-
-    if (lookUpElement == localElement) {
+    if (lookUpElement == localElement || lookUpElement == null) {
```

Comment:
Please add a comment indicating the conditions under which `lookUpElement` will be `null`. Given that we resolved it to something with the `this` (the `localElement` is not `null`), I can't think of a case where look-up would fail.